src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -1004,6 +1034,28 @@ public override void Accept(OperationVisitor visitor)
         }
     }
 
+    internal sealed partial class BoundDeconstructionAssignmentOperator : BoundExpression
+    {
+        protected override OperationKind ExpressionKind => OperationKind.None;
+
+        public override void Accept(OperationVisitor visitor)
+        {
+            visitor.VisitNoneOperation(this);
+        }
+
+        public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
+        {
+            return visitor.VisitNoneOperation(this, argument);
+        }
+
+        internal class AssignmentInfo
+        {
+            public BoundAssignmentOperator Assignment;
+            public BoundLValuePlaceholder LValuePlaceholder;
+            public BoundRValuePlaceholder RValuePlaceholder;
+        }
```

Comment:
Makes sense. Moved into `BoundExpression.cs`.
I've been wondering whether this should be a struct. What do you think?